tui/offer.py

```diff
@@ -0,0 +1,145 @@
+from dataclasses import dataclass, asdict
+from typing import Optional, Tuple
+import asyncio
+import hashlib
+import urllib.request
+import urllib.parse
+from pathlib import Path
+import os
+
+from common import accept_cookies, query_accordion
+
+from pyppeteer.page import Page
+from pyppeteer.element_handle import ElementHandle
+
+
+@dataclass
+class Data:
+    @dataclass
+    class Room:
+        is_apartament: Optional[bool] = None
+        is_studio: Optional[bool] = None
+        size: Optional[str] = None
+
+    @dataclass
+    class Transport:
+        organised: Optional[bool] = None
+        type: Optional[str] = None
+
+    operator: Optional[str] = None
+    hotel: Optional[str] = None
+    country: Optional[str] = None
+    city: Optional[str] = None
+    description: Optional[str] = None
+    score: Optional[int] = None
+    kids: Optional[bool] = None
+    url: Optional[str] = None
+    image: Optional[str] = None
+    image_local: Optional[str] = None
+
+    room: Optional[Room] = None
+    transport: Optional[Transport] = None
+
+
+async def _get_rating(page: Page) -> int:
+    rating_wrapper = await page.querySelector(".top-section__hotel-rating")
+    assert rating_wrapper is not None
+    return len(await rating_wrapper.querySelectorAll("li"))
+
+
+async def _get_hotel_name(page: Page) -> str:
+    header = await page.querySelector(".top-section__hotel-name")
+    assert header is not None
+    return await (await header.getProperty("textContent")).jsonValue()
+
+
+async def _get_location(page: Page) -> Tuple[str, str]:
```

Comment:
jeżeli korzystasz z pythona > 3.9 to możesz wykorzystać wbuodwane `tuple` zamiast tego z `typing`, to z typingu od wersji 3.9 jest uznawane za deprecated

Reply:
Nie wiedziałem, ale mega dobrze wiedzieć